File: .framework-tests/java-testng/src/test/java/com/saucedemo/testng/CartTests.java
package com.saucedemo.testng;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.Assert;
import org.testng.annotations.*;

public class CartTests {

    private WebDriver driver;

    @BeforeMethod
    public void setUp() {
        driver = new ChromeDriver();
        driver.get("https://www.saucedemo.com");
        driver.findElement(By.id("user-name")).sendKeys("standard_user");
        driver.findElement(By.id("password")).sendKeys("secret_sauce");
        driver.findElement(By.id("login-button")).click();
    }

    @AfterMethod
    public void tearDown() {
        driver.quit();
    }

    /**
     * Adding a single item updates the cart badge to 1.
     * 1. Log in as standard_user and navigate to the inventory page
     * 2. Click add-to-cart for Sauce Labs Backpack
     * 3. Check: shopping cart badge text equals "1"
     */
    // @tc:32984
    @Test(groups = {"smoke"})
    public void addSingleItem_UpdatesBadgeToOne() {
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).click();
        String badge = driver.findElement(By.className("shopping_cart_badge")).getText();
        Assert.assertEquals(badge, "1", "Cart badge should show 1 after adding one item");
    }

    /**
     * Adding three items shows count 3 on the cart badge.
     * 1. Log in and navigate to the inventory page
     * 2. Add Sauce Labs Backpack, Bike Light, and Bolt T-Shirt to cart
     * 3. Check: shopping cart badge text equals "3"
     */
    // @tc:32985
    @Test
    public void addThreeItems_BadgeShowsThree() {
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).click();
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-bike-light']")).click();
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-bolt-t-shirt']")).click();
        String badge = driver.findElement(By.className("shopping_cart_badge")).getText();
        Assert.assertEquals(badge, "3");
    }

    /**
     * Cart page lists all added items.
     * 1. Log in and add Sauce Labs Backpack and Bike Light to cart
     * 2. Click the shopping cart icon
     * 3. Check: cart page shows exactly 2 cart items
     */
    // @tc:32986
    @Test
    public void cartPage_ListsAddedItems() {
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).click();
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-bike-light']")).click();
        driver.findElement(By.className("shopping_cart_link")).click();
        int cartItemCount = driver.findElements(By.className("cart_item")).size();
        Assert.assertEquals(cartItemCount, 2, "Cart should contain 2 items");
    }
}

File: .framework-tests/csharp-mstest/Tests/CartTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;

namespace SauceDemo.MSTest.Tests
{
    [TestClass]
    public class CartTests
    {
        private IWebDriver _driver;

        [TestInitialize]
        public void Setup()
        {
            _driver = new ChromeDriver();
            _driver.Navigate().GoToUrl("https://www.saucedemo.com");
            _driver.FindElement(By.Id("user-name")).SendKeys("standard_user");
            _driver.FindElement(By.Id("password")).SendKeys("secret_sauce");
            _driver.FindElement(By.Id("login-button")).Click();
        }

        [TestCleanup]
        public void Teardown() => _driver.Quit();

        /// <summary>
        /// Adding an item updates the cart badge to 1.
        /// 1. Log in as standard_user and navigate to the inventory page
        /// 2. Click add-to-cart for Sauce Labs Backpack
        /// 3. Check: shopping cart badge text equals "1"
        /// </summary>
        [TestMethod]
        [TestProperty("tc", "32963")]
        [TestCategory("smoke")]
        public void AddItem_UpdatesCartBadge()
        {
            _driver.FindElement(By.CssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).Click();
            var badge = _driver.FindElement(By.ClassName("shopping_cart_badge")).Text;
            Assert.AreEqual("1", badge);
        }

        /// <summary>
        /// Removing one item from two decrements the cart badge to 1.
        /// 1. Add Sauce Labs Backpack and Bike Light to cart
        /// 2. Click Remove for Sauce Labs Bike Light
        /// 3. Check: shopping cart badge text equals "1"
        /// </summary>
        [TestMethod]
        [TestProperty("tc", "32964")]
        public void RemoveItem_DecrementsCartBadge()
        {
            _driver.FindElement(By.CssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).Click();
            _driver.FindElement(By.CssSelector("[data-test='add-to-cart-sauce-labs-bike-light']")).Click();
            _driver.FindElement(By.CssSelector("[data-test='remove-sauce-labs-bike-light']")).Click();
            var badge = _driver.FindElement(By.ClassName("shopping_cart_badge")).Text;
            Assert.AreEqual("1", badge);
        }

        /// <summary>
        /// Cart page shows the correct price for the added item.
        /// 1. Add Sauce Labs Backpack to cart
        /// 2. Click the shopping cart icon to open the cart page
        /// 3. Check: item price displayed equals "$29.99"
        /// </summary>
        [TestMethod]
        [TestProperty("tc", "32965")]
        public void CartPage_ShowsCorrectItemPrice()
        {
            _driver.FindElement(By.CssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).Click();
            _driver.FindElement(By.ClassName("shopping_cart_link")).Click();
            var price = _driver.FindElement(By.ClassName("inventory_item_price")).Text;
            Assert.AreEqual("$29.99", price);
        }
    }
}
